client/src/js/samples/components/LabelEditor/ColorSelector.js

```diff
@@ -0,0 +1,94 @@
+import React from "react";
+import styled from "styled-components";
+import { Button, Input, InputGroup, InputLabel, InputError } from "../../../base";
+
+export const StyledColorButton = styled(Button)`
+    background-color: ${props => props.color};
+    margin: 0px 2px;
+`;
+
+export const StyledInput = styled(Input)`
+    width: 40%;
+`;
+
+const getInitialState = ({ color }) => ({
+    color: color || ""
+});
+
+export class ColorSelector extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = getInitialState(this.props);
```

Comment:
You are managing state here and in your parent component.

Keep the state in `CreateLabel`. The `ColorSelector` component should take the color value as a prop and pass events back to `CreateLabel` using your `onColorChange` prop.